#!/bin/bash

mkdir /home/aazzolini/gdebiase/hs_T2Tv2/
for run in $(ls /home/aazzolini/gdebiase/fastp); do
    mkdir /home/aazzolini/gdebiase/hs_T2Tv2/${run}_STAR_mapping_result 
    for seq in $(ls /home/aazzolini/gdebiase/fastp/${run} | grep _R1_ | cut -d R -f 1 | uniq); do
        conda run -n al STAR \
        --runThreadN 40 \
        --genomeDir /home/aazzolini/gdebiase/hs_T2T/hs_T2T_genomeindex \
        --readFilesCommand zcat \
        --readFilesIn /home/aazzolini/gdebiase/fastp/${run}/${seq}R1_trimmed.fastq.gz /home/aazzolini/gdebiase/fastp/${run}/${seq}R2_trimmed.fastq.gz \
        --alignEndsType EndToEnd \
        --outSAMattributes Standard \
        --outFilterMismatchNoverReadLmax 0.04 \
        --outReadsUnmapped Fastx \
        --outSAMprimaryFlag AllBestScore \
        --outFilterType BySJout \
        --outSAMtype BAM SortedByCoordinate \
        --chimOutType WithinBAM HardClip \
        --alignSJoverhangMin 6 \
        --alignSJDBoverhangMin 6 \
	--sjdbOverhang 65 \
        --outFileNamePrefix /home/aazzolini/gdebiase/hs_T2Tv2/${run}_STAR_mapping_result/${seq}
    done
done


#--clip5pNbases 5 \
#--clip3pNbases 5 \
#--alignEndType: type of read ends alignment, EndtoEnd: force end-to-end read alignment, do not soft-clip, p.42
#--outSAMattributes: a string of desired SAM attributes, in the order desired for the output, Standard (dafault): NH HI AS nM
        #https://en.wikipedia.org/wiki/SAM_(file_format)
        #NH: Number of reported alignments that contain the query in the current record
        #HI: Query hit index
        #AS: Alignment score generated by aligner
        #NM: Edit distance to the reference
#--outFilterMismatchNoverReadLmax: real: alignment will be output only if its ratio of mismatches to *read* length is
        #less than or equal to this value. p.30
#--outReadsUnmapped: output of unmapped and partially mapped (i.e. mapped only one mate
        #of a paired end read) reads in separate file(s).; Fastx: output in separate fasta/fastq files, Unmapped.out.mate1/2 p.28
#--outSAMprimaryFlag: which alignments are considered primary - all others will be marked
        #with 0x100 bit in the FLAG; AllBestScore: all alignments with the best score are primary p.31
#--outFilterType: type of filtering; BySJout keep only those reads that contain junctions that passed filtering into SJ.out.tab p.8
#--chimOutType: type of chimeric output; WithinBAM HardClip (default) hard-clipping in the CIGAR for supplemental chimeric
        #alignments (default if no 2nd word is present) p.44
#--alignSJoverhangMin: minimum overhang for unannotated junctions; 6 p.8
#--alignSJDBoverhangMin: minimum overhang for annotated junctions; 6 p.8
#--outFileNamePrefix: STAR produces multiple output files. All files have standard name, however, you can change the file
        #prefixes using --outFileNamePrefix /path/to/output/dir/prefix. By default, this parameter is
        #./, i.e. all outputfiles are written in the current directory.
